### PR TITLE
Sundhar dev

### PR DESCRIPTION
**Pull request by sundharm**
[AB#100](https://dev.azure.com/Tringo/cabcbfb6-3a22-44cf-ac98-b3810925a870/_workitems/edit/100)
Added toggling to select between known and unknown dates

### CODE REVIEW

File: app/src/components/searchWidget/date-input/expandedDatePanel.tsx [hunk 1]
```diff
@@ -79,24 +83,33 @@ export class ExpandedDatePanel extends React.Component<StateChangedProps, any> {
...
         return (
             <div className="date-panel-expanded">
-                <div className="top-toogler">
+                <ToggleButtonGroup className="datetoggle" name="options" defaultValue={1}>
                     <div
                         className="dates-selector middle-text"
                         onClick={() => this.selectPanel(datePanelTypes.SPECIFIC_DATES)}
                     >
-                        <div>Specific Dates</div>
+                        <Button onClick={this.changeColor.bind(this)}>Specific Dates</Button>
```
Comment (s3419069) [inline app/src/components/searchWidget/date-input/expandedDatePanel.tsx]: Please move this onClick actions to .selectPanel() method

Comment (s3419069) [reply app/src/components/searchWidget/date-input/expandedDatePanel.tsx]: Selected button should also keep selected after you select-close-open that panel. doesn't look like it's covered

Comment (s3419069) [reply app/src/components/searchWidget/date-input/expandedDatePanel.tsx]: maybe set color for a button in .saveSelectedPanel()

Comment (sundharm) [reply app/src/components/searchWidget/date-input/expandedDatePanel.tsx]: yes I'm trying to set it in saveSelectedPanel()

**Comment by rmit-s3668804-Dzmitry-Kakaruk** [comment]
Doesn't look like it should be

image

File: app/src/components/searchWidget/date-input/expandedDatePanel.tsx (hunk 1 above)
Comment (sundharm) [reply app/src/components/searchWidget/date-input/expandedDatePanel.tsx]: @rmit-s3668804-Dzmitry-Kakaruk I could not figure out what exact type of buttons is being used for this part by webjet. Any leads about that would be helpful.

Comment (sundharm) [reply app/src/components/searchWidget/date-input/expandedDatePanel.tsx]: @rmit-s3668804-Dzmitry-Kakaruk and @mmo-mikhail Fixed it to look as close as webjet's and the state now works.

File: app/src/components/searchWidget/date-input/expandedDatePanel.tsx [hunk 2]
```diff
@@ -66,37 +71,53 @@ export class ExpandedDatePanel extends React.Component<StateChangedProps, any> {
...
             <div className="date-panel-expanded">
-                <div className="top-toogler">
+                <ToggleButtonGroup className="datetoggle" name="options" defaultValue={1}>
                     <div
                         className="dates-selector middle-text"
                         onClick={() => this.selectPanel(datePanelTypes.SPECIFIC_DATES)}
                     >
-                        <div>Specific Dates</div>
+                        <Button
+                            active={this.props.initialModel.clicked}
```
Comment (s3419069) [inline app/src/components/searchWidget/date-input/expandedDatePanel.tsx]: active={this.props.initialModel.unknownDates}

File: app/src/components/searchWidget/date-input/expandedDatePanel.tsx [hunk 3]
```diff
@@ -66,37 +71,53 @@ export class ExpandedDatePanel extends React.Component<StateChangedProps, any> {
...
+                            Specific Dates
+                        </Button>
                     </div>
                     <div
                         className="dates-selector middle-text"
                         onClick={() => this.selectPanel(datePanelTypes.UNKNOWN_DATES)}
                     >
-                        <div>Flexible Dates</div>
+                        <Button
+                            active={this.props.initialModel.clickedu}
```
Comment (s3419069) [inline app/src/components/searchWidget/date-input/expandedDatePanel.tsx]: active={!this.props.initialModel.unknownDates}

so REMOVE clicked/clickedu states

**Review by s3419069 — APPROVED** [review]

**Pull request merged**